src/customHooks/useFetch.js

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from 'react';
+
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    const abortCont = new AbortController();
```

Comment:
oou what does this do?

Reply:
This cancels the fetch request if the user were to navigate away or something before it could get a response. Prevents getting an error.